src/plugins/intel_cpu/tests/unit/shape_inference_test/eye_shape_inference_test.cpp

```diff
@@ -0,0 +1,171 @@
+                                        {3, std::make_shared<HostTensor>(element::i32, Shape{3}, batch.data())}};
+
+    input_shapes = ShapeVector{{}, {1}, {1}, {3}};
+
+    OV_EXPECT_THROW(shape_inference(op.get(), input_shapes, output_shapes, const_data),
+                    AssertFailure,
+                    HasSubstr("Value -8 not in range [0:"));
+}
+
+TEST_F(EyeV9StaticShapeInferenceTest, assert_on_rows_not_1D) {
+    const auto rows_node = std::make_shared<Parameter>(element::i64, PartialShape::dynamic());
+    const auto cols_node = std::make_shared<Parameter>(element::i64, PartialShape::dynamic());
+    const auto diag_node = std::make_shared<Parameter>(element::i32, PartialShape::dynamic());
+    const auto batch_node = std::make_shared<Parameter>(element::i32, PartialShape::dynamic());
+
+    const auto op = make_op(rows_node, cols_node, diag_node, batch_node, element::i16);
+
+    int64_t cols = 8;
+    auto rows = std::array<int64_t, 2>{2, 1};
+    auto batch = std::array<int32_t, 3>{2, 4, 1};
+    const auto const_data = std::map<size_t, HostTensorPtr>{
+        {0, std::make_shared<HostTensor>(element::i64, Shape{rows.size()}, rows.data())},
+        {1, std::make_shared<HostTensor>(element::i64, Shape{1}, &cols)},
+        {3, std::make_shared<HostTensor>(element::i32, Shape{batch.size()}, batch.data())}};
+
+    input_shapes = ShapeVector{{}, {1}, {1}, {3}};
+
+    OV_EXPECT_THROW(shape_inference(op.get(), input_shapes, output_shapes, const_data),
+                    NodeValidationFailure,
+                    HasSubstr("'num_rows' value must be a scalar or 1D tensor. Got:"));
+}
+
+TEST_F(EyeV9StaticShapeInferenceTest, assert_on_columns_not_1D) {
+    const auto rows_node = std::make_shared<Parameter>(element::i64, PartialShape::dynamic());
+    const auto cols_node = std::make_shared<Parameter>(element::i64, PartialShape::dynamic());
+    const auto diag_node = std::make_shared<Parameter>(element::i32, PartialShape::dynamic());
+    const auto batch_node = std::make_shared<Parameter>(element::i32, PartialShape::dynamic());
+
+    const auto op = make_op(rows_node, cols_node, diag_node, batch_node, element::i16);
+
+    int64_t rows = 8;
+    auto cols = std::array<int64_t, 2>{2, 1};
+    auto batch = std::array<int32_t, 3>{2, 4, 1};
+    const auto const_data = std::map<size_t, HostTensorPtr>{
+        {0, std::make_shared<HostTensor>(element::i64, Shape{}, &rows)},
+        {1, std::make_shared<HostTensor>(element::i64, Shape{cols.size()}, cols.data())},
+        {3, std::make_shared<HostTensor>(element::i32, Shape{batch.size()}, batch.data())}};
+
+    input_shapes = ShapeVector{{1}, {}, {1}, {3}};
+
+    OV_EXPECT_THROW(shape_inference(op.get(), input_shapes, output_shapes, const_data),
+                    NodeValidationFailure,
+                    HasSubstr("'num_columns' value must be a scalar or 1D tensor. Got:"));
+}
+
+TEST_F(EyeV9StaticShapeInferenceTest, assert_on_batch_shape_not_match_shape_in_const_data) {
```

Comment:
Additional test with default constructor and shape_infer could be added to ensure the inputs are not required to be set.

Reply:
Do you  mean test like:
https://github.com/openvinotoolkit/openvino/blob/c1200e2a6347fc10af37afcc48dbd660253f8d8a/src/core/tests/type_prop/eye.cpp#L360

I've put this kind test in type prop for default ctor also for some other ops.

Reply:
Exactly, okay.